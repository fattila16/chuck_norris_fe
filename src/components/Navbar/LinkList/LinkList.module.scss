@import 'src/assets/scss/color-palette';
@import 'src/assets/scss/vars';

.ul {
    display: flex;
    margin: 0;
    padding: 0;

    li {
        list-style-type: none;
        
        a {
            padding: 0px 10px;
            color: map-get($active-theme, silk );
            text-decoration: none;
        }
    }
}