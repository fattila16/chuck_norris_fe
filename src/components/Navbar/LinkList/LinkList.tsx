import React from 'react';
import Link from 'next/link';
import css from './LinkList.module.scss';

export interface NavLink {
    name: string;
    href: string;
}

interface LinkListProps {
    links: NavLink[];
}

const LinkList: React.FC<LinkListProps> = ({ links }) => {
    return (
        <ul className={css.ul}>
            {links.map((link) => (
                <li key={link.href}>
                    <Link href={link.href}>
                        <a href={link.href}>{link.name}</a>
                    </Link>
                </li>
            ))}
        </ul>
    );
};

export default LinkList;
