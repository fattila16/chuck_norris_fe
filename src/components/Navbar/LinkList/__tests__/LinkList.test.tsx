import React from 'react';
import { shallow } from 'enzyme';
import LinkList from '../LinkList';

describe('[LinkList]', () => {
    it('Should render LinkList component without error', () => {
        const component = shallow(<LinkList links={[{ name: '/', href: '/' }]} />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
