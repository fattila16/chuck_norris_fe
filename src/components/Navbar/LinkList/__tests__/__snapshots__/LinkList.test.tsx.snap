// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[LinkList] Should render LinkList component without error 1`] = `
Array [
  <ul
    className="ul"
  >
    <li>
      <Link
        href="/"
      >
        <a
          href="/"
        >
          /
        </a>
      </Link>
    </li>
  </ul>,
]
`;
